tools/cli/commands/changelog.js

```diff
@@ -448,13 +451,17 @@ async function changedProjects() {
 	const projects = allProjects();
 	// Get all files that were worked with (created, deleted, modified, etc)
 	for ( const file of gitStatus.files ) {
-		gitFiles.push( file.path );
+		const parsedPath = file.path.split( '/' );
+		const project = `${ parsedPath[ 1 ] }/${ parsedPath[ 2 ] }`;
+		if ( ! gitFiles.includes( project ) ) {
+			gitFiles.push( project );
+		}
 	}
 
 	// See if any files modified match our project list.
 	for ( const proj of projects ) {
 		for ( const file of gitFiles ) {
-			if ( file.includes( proj ) && ! modifiedProjects.includes( proj ) ) {
+			if ( proj === file ) {
```

Comment:
`file` isn't really a file anymore.

GH won't let me make it a suggestion, but something like this might be a little more straightforward:

```js
const re = /^projects\/([^\/]+\/[^\/]+)\//;
const modifiedProjects = new Set();
const git = simpleGit();
const gitStatus = await git.status();
for ( const file of gitStatus.files ) {
    const m = file.path.match( re );
    if ( m ) {
        modifiedProjects.add( m[1] );
    }
}
return allProjects().filter( v => modifiedProjects.has( v ) );
```

Reply:
Nice, thanks for the suggestion! Learned a lot about regex and TIL about Sets. Updated 👍